### DISCUSSION TITLE
Getting 502 Bad Gateway if I try to shorten an existing URL through API

### DISCUSSION THREAD

**Discussion by chromaniac2** [YOURLS/YOURLS#3318] [Q&A] [answered]
This started happening after upgrading to YOURLS 1.9. 

I use a Telegram bot to create short links. And after upgrading to latest update, I realized that I was getting a lot of errors. Ended up using API URL to check the issue directly in browser and realized that the server end up showing 502 Bad Gateway error if I try to reload the API URL to create a short link. It works fine on a fresh link. It works fine on Yourls admin panel. 

Basically if I run this: 

`https://myyourlsdomain.com/yourls-api.php?signature=XYZ&action=shorturl&url=http://domain.com/brandnewurl`

It works fine. But if I run the URL again (now that it is added to database... It gives a 502 Bad Gateway error.

**Comment by ozh** [comment]
That behavior isn't produced by YOURLS itself - plugin maybe ?

Shortening twice gives you the following : 

```json
{"status":"fail",
  "code":"error:url",
  "message":"http:\/\/lol.com\/% already exists in database (short URL: 127.0.0.1\/ozh.in\/j)",
  "errorCode":"",
  "statusCode":"",
  "url":{
    "keyword":"j",
    "url":"http:\/\/lol.com\/%",
    "title":"LOL.com - Premium category defining domain names for sale, exact match domains",
    "date":"2022-05-02 20:37:19",
    "ip":"127.0.0.1",
    "clicks":0
  }
  "title":"LOL.com - Premium category defining domain names for sale, exact match domains",
  "shorturl":"https:\/\/127.0.0.1\/ozh.in\/j"
}
```

... with a `500 Internal Server Error`, which isn't too great, I'll check this -- #3319

**Reply by chromaniac2** [reply]
Yeah, I did try after disabling all addons. I had a feeling that this addon (https://github.com/adigitalife/yourls-change-error-messages) might be causing this issue. But disabling did not change the behavior. I also disabled Cloudflare which switched the Cloudflare 502 error message to Nginx 502 error message. I even ended up restoring the folder backup from before 1.9 update but even that did not change the behavior (I did not restore database though). I am out of ideas right now. Would keep an eye on #3319 thanks! 

For now, I am just going to use `define( 'YOURLS_UNIQUE_URLS', false );` to get around this issue. 

This seems to be the error from the logs. 

> 2022/05/02 21:14:41 [error] 22859#22859: *753052 upstream sent invalid status "0" while reading response header from upstream, client: 162.158.162.224, server: DOMAIN.COM, request: "GET /yourls-api.php?signature=REMOVED&action=shorturl&url=google.com HTTP/1.1", upstream: "fastcgi://unix:/var/run/php/php7.4-fpm.sock:", host: "DOMAIN.COM"

**Comment by chromaniac2** [comment, marked as answer]
1.9.1 update seems to have fixed this issue for me. Thanks!
